rust/src/types/execution_context.rs

```diff
@@ -0,0 +1,121 @@
+use evmc_vm::{
+    AccessStatus, Address, ExecutionContext, ExecutionMessage, ExecutionResult, ExecutionTxContext,
+    StorageStatus, Uint256,
+};
+
+#[cfg_attr(test, mockall::automock)]
+pub trait ExecutionContextTrait {
```

Comment:
I am not a big fan of Naming types by its type. 
This is an execution context. The derived types are execution context instances, each with its own flavor. 

-  `MockExecutionContextTrait` is not a good name 
- the old `ExecutionContext<'a>` should be `EvmcExectutionContext<'a>`

Reply:
ExecutionContext is a type from evmc. The naming of ExecutionContextTrait is indeed not ideal but at the end it is a trait with the same methods as the struct ExecutionContext.

Reply:
i see :(